iwyu_regex.cc

```diff
@@ -46,4 +46,31 @@ bool RegexMatch(RegexDialect dialect, const std::string& str,
   CHECK_UNREACHABLE_("Unexpected regex dialect");
 }
 
+// Returns the result of replacing the first match of the `pattern` regular
+// expression in `str` with `replacement`. `dialect` specifies what regular
+// expression dialect to use.
+std::string RegexReplace(
```

Comment:
Formatting looks a little off here -- could you use clang-format on the new code? (I also have to recommend https://github.com/kimgr/git-format/, because that's my favorite way to clean up patch sets :smile:)

Reply:
My favorite way is to hit a shortcut to invoke clang-format in my IDE ;)
It looks like lots of code isn't correctly formatted in this file. Perhaps a batch formatting of the code base would be useful. Unless my clang-format isn't formatting as it should?

Reply:
That's why I prefer to do it patch-hunk-by-patch-hunk instead of file-by-file. The Emacs clang-format plugin does region formatting, so I typically just reformat what I just changed, and then use `git-format` to check/fix-up individual commits.

But yeah, at some point maybe we should take the plunge and reformat everything. I just don't like such big changes :)